src/main/java/com/kodilla/ecommercee/domain/Cart.java

```diff
@@ -1,4 +1,63 @@
 package com.kodilla.ecommercee.domain;
 
+import com.kodilla.ecommercee.exception.ProductNotFoundInCartException;
+
+import javax.persistence.Entity;
+import javax.persistence.Table;
+import javax.persistence.GeneratedValue;
+import javax.persistence.ManyToMany;
+import javax.persistence.GenerationType;
+import javax.persistence.Id;
+import javax.persistence.CascadeType;
+import javax.persistence.JoinTable;
+import javax.persistence.JoinColumn;
+import javax.validation.constraints.NotNull;
+import java.util.ArrayList;
+import java.util.List;
+
+@Entity
+@Table(name="CARTS")
 public class Cart {
+    private Long id;
+    private List<Product> products = new ArrayList<>();
+
+    public Cart() {};
+
+    public Cart(Long id, List<Product> productList) {
+        this.id = id;
+        this.products = productList;
+    }
+
+    @Id
+    @NotNull
+    @GeneratedValue(strategy = GenerationType.AUTO)
+    public Long getId() {
+        return id;
+    }
+
+    @ManyToMany(cascade = CascadeType.ALL)
+    @JoinTable(
+            name = "JOIN_CART_PRODUCT",
+            joinColumns = {@JoinColumn(name = "PRODUCT_ID", referencedColumnName = "PRODUCT_ID")},
+            inverseJoinColumns = {@JoinColumn(name = "CART_ID", referencedColumnName = "CART_ID")}
+    )
+    public List<Product> getProductList() {
+        return products;
+    }
+
+    public void addToCart(Product product) {
+        products.add(product);
+    }
+
```

Comment:
Kacper czy aby na pewno powinnismy definiować tutaj metody? aż musialem sie do modułu cofnąć... wydaje mi sie że te zadania wykonuje klasa service i repository..

Reply:
Ale nawet, jesli będzie do tego metoda w service i repository, to będzie potrzebny dostęp do prywatnej zmiennej jaką jest lista produktów. Więc pomyślałem o bezpośredniej metodzie dodania i usunięcia produktów tak, żeby potem nie powstawały tasiemce w stylu getList().addToList()

Reply:
Rzeczywiście, aż sam też się wróciłem. Trochę się rozpędziłem

Reply:
popieram @przemyslawkrawczynski . w mojej ocenie również to nie powinno być tutaj.